blog/utils/handlePosts.ts

```diff
@@ -71,13 +85,19 @@ export const slicePosts = (
  * @param posts Posts to be handled
  * @param sortBy Sort option (must be: "date_desc" | "date_asc" | "title_asc" | "title_desc" )
  * @param slug Category slug to be filter
+ * @param term Term to be filter
  */
 export default function handlePosts(
   posts: BlogPost[],
   sortBy: SortBy,
   slug?: string,
+  term?: string,
 ) {
-  const filteredPosts = filterPostsByCategory(posts, slug);
+  const filteredPosts = slug
+    ? filterPostsByCategory(posts, slug)
+    : term
+    ? filterPostsByTerm(posts, term)
+    : posts;
```

Comment:
its not a "dont" if you prefer dont do it, but i think it's more legible make a function `filterPost` with early return and call here.

Somethig like this:
```
export const filterPosts =  ({posts, term, slug}) => {
 if(!term && !slug) return posts
 if(term) return filterPostsByTerm(posts, term)
 return filterPostsByCategory(posts, slug)
} 
```